src/shared/varlink-io.systemd.service.c

```diff
@@ -27,3 +27,29 @@ int varlink_method_ping(Varlink *link, JsonVariant *parameters, VarlinkMethodFla
 
         return varlink_reply(link, NULL);
 }
+
+int varlink_method_set_log_level(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                {"level", JSON_VARIANT_INTEGER, json_dispatch_int64, 0},
+                {}
+        };
+
+        int64_t level;
+        int r;
+
+        assert(link);
+        assert(parameters);
+
+        if (json_variant_elements(parameters) != 2)
```

Comment:
2?

Reply:
Blah. I just remembered why did I use `2` in the udev function I copied this from... The `parameters` _does_ actually contain 2 elements, because the key and value of a pair are stored separately in a `JsonObject`. I.e.,

```c
_cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
   
json_build(&v, JSON_BUILD_OBJECT(JSON_BUILD_PAIR("foo", JSON_BUILD_CONST_STRING("bar"))));

/* Expected */ 
assert_se(streq(json_variant_string(json_variant_by_key(v, "foo")), "bar"));

/* Unexpected */
assert_se(json_variant_elements(v) == 2);
assert_se(streq(json_variant_string(json_variant_by_index(v, 0)), "foo"));
assert_se(streq(json_variant_string(json_variant_by_index(v, 1)), "bar"));
```